### ISSUE TITLE
Function sortBy throws MixedTypeError if value in sorted column is null

### ISSUE THREAD

**Issue by ivanlucky22** [Gmousse/dataframe-js#62]
Hi team,

I'm using dataframe-js v1.3.2 and facing an error when trying to sort dataframe by column that contains null. Please see the following example:
```
let result = new DataFrame([
  {Name: 'Peter', Age: 16},
  {Name: 'Denis', Age: null}
], ['Name', 'Age']);

result.sortBy('Age')
```
Please let me know if you are going to fix this in the nearest release. Or please share some thoughts on a workaround. 
PS. in the project value really needs to stay null so please do not propose change null to zero :)

**Comment by Gmousse** [comment]
Hi ! 
Thank you for using dataframe-js.

>  PS. in the project value really needs to stay null so please do not propose change null to zero :)

Ahah don't worry :d .

> facing an error when trying to sort dataframe by column that contains null

The reason why I have introduced the MixedTypeError (#6 #3) is due to some strange behaviors when we compare something to another type of values (which generally brokes the sorted results).

Generally, it's better to handle only one data type by column. 

I will see to introduce an alternative method (like .dangerousSortBy) to handle this kind of things if you can't convert your data.

**Comment by episodeyang** [comment]
@Gmousse guess in this case, all of the scalar types are intact "nullable"s, scalar types that can also be `null`.

**Comment by episodeyang** [comment]
@Gmousse while you are at it, whether we place the `NaN` values at the *beginning* or the *end* of the sorted DataFrame depends on the user. I was wondering whether this warrants an extra argument, and apparently it does. See `pandas` documentation here:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html

```python
>>> df.sort_values(by='col1', ascending=False, na_position='first')
    col1 col2 col3
3   NaN  8    4
4   D    7    2
5   C    4    3
2   B    9    9
0   A    2    0
1   A    1    1
```

**Comment by Gmousse** [comment]
Hi @episodeyang,
Indeed it's a good idea. I will see to add an additionnal parameter in order to place null, undefined or NaN values first or last.

I will try it today. I hope a release on the 1.4.0.

Stay tuned.

**Comment by episodeyang** [comment]
Thanks Gmousse! On type:

just an inspiration: in flow (and in graphQL in this case), nullables are different from their non-nullable counter parts. For example in this error: I was passing in a variable `$metricFiles: [string]` into a field `metricFiles: [string]!`, and got the following error (pardon the brevity of the description). 

This is because the type `[string]` and `[string]!` are different in the graphQL type checking.

```javascript
of type "[String]" used in position expecting type "[String]!
```

**Comment by Gmousse** [comment]
@episodeyang Im working on something like this: 

```
sortBy(columnNames, reverse = false, nullsPosition = 'first')
sortBy(columnNames, reverse = false, nullsPosition = 'last')
```

The algorithm:
- If the value is in undefined, null or NaN => place at first or last (depending on the parameter)
- else if the value is not the same type than previously (and not undefined, null or NaN), throw MixedTypeError
- else the behaviour is the standard sort algorithm

**Comment by Gmousse** [comment]
Hi @episodeyang,

It's resolved (the code is on develop) and it will be released in the 1.4.0.

**Comment by Gmousse** [comment]
Released in 1.4.0
